compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/FirExpressionsResolveTransformer.kt

```diff
@@ -1115,41 +1108,34 @@ open class FirExpressionsResolveTransformer(transformer: FirAbstractBodyResolveT
             })
         }
 
-        dataFlowAnalyzer.enterCall()
-        var callCompleted = true
-        var result = delegatedConstructorCall
-        try {
-            val lastDispatchReceiver = implicitReceiverStack.lastDispatchReceiver()
-            context.forDelegatedConstructorCall(containingConstructor, containingClass as? FirRegularClass, components) {
-                delegatedConstructorCall.transformChildren(transformer, ResolutionMode.ContextDependent)
-            }
+        dataFlowAnalyzer.enterDelegatedConstructorCall()
+        val lastDispatchReceiver = implicitReceiverStack.lastDispatchReceiver()
```

Comment:
Originally this `try/finally` was needed not for cases when there is an expression during resolution, but just for removal of need to call `dataFlowAnalyzer.exitDelegatedConstructorCall` on each return in this function (there were several returns some time ago)